pkg/deploy/api/validation/validation.go

```diff
@@ -255,29 +257,18 @@ func validateImageChangeParams(params *deployapi.DeploymentTriggerImageChangePar
 	errs := fielderrors.ValidationErrorList{}
 
 	if len(params.From.Name) != 0 {
-		if len(params.From.Kind) == 0 {
-			params.From.Kind = "ImageStream"
-		}
 		kinds := sets.NewString("ImageRepository", "ImageStream", "ImageStreamTag")
 		if !kinds.Has(params.From.Kind) {
-			msg := fmt.Sprintf("kind must be one of: %s", strings.Join(kinds.List(), ", "))
-			errs = append(errs, fielderrors.NewFieldInvalid("from.kind", params.From.Kind, msg))
+			errs = append(errs, fielderrors.NewFieldInvalid("from.kind", params.From.Kind, "kind must be an ImageStreamTag"))
 		}
-
-		if !kvalidation.IsDNS1123Subdomain(params.From.Name) {
-			errs = append(errs, fielderrors.NewFieldInvalid("from.name", params.From.Name, "name must be a valid subdomain"))
+		if err := validateImageStreamTagName(params.From.Name); err != nil {
+			errs = append(errs, fielderrors.NewFieldInvalid("from.name", params.From.Name, err.Error()))
 		}
 		if len(params.From.Namespace) != 0 && !kvalidation.IsDNS1123Subdomain(params.From.Namespace) {
 			errs = append(errs, fielderrors.NewFieldInvalid("from.namespace", params.From.Namespace, "namespace must be a valid subdomain"))
 		}
-
-		if len(params.RepositoryName) != 0 {
-			errs = append(errs, fielderrors.NewFieldInvalid("repositoryName", params.RepositoryName, "only one of 'from', 'repository' name may be specified"))
-		}
 	} else {
-		if len(params.RepositoryName) == 0 {
-			errs = append(errs, fielderrors.NewFieldRequired("from"))
-		}
+		errs = append(errs, fielderrors.NewFieldRequired("from"))
```

Comment:
Nit: Can you reverse the logic in the `if` so the `if` and `else` are very close together?